dbt/adapters/duckdb/plugins/delta.py

```diff
@@ -43,6 +44,110 @@ def load(self, source_config: SourceConfig):
     def default_materialization(self):
         return "view"
 
+    def store(self, target_config: TargetConfig, df=None):
+        mode = target_config.config.get("mode", "overwrite")
+        table_path = target_config.location.path
+        storage_options = target_config.config.get("storage_options", {})
+
+        if mode == "overwrite_partition":
+            partition_key = target_config.config.get("partition_key", None)
+            if not partition_key:
+                raise Exception(
+                    "'partition_key' has to be defined when mode 'overwrite_partition'!"
+                )
+
+            if isinstance(partition_key, str):
+                partition_key = [partition_key]
+
+            partition_dict = []
+            for each_key in partition_key:
+                unique_key_array = pc.unique(df[each_key])
+
+                if len(unique_key_array) == 1:
+                    partition_dict.append((each_key, str(unique_key_array[0])))
+                else:
+                    raise Exception(
+                        f"'{each_key}' column has not one unique value, values are: {str(unique_key_array)}"
+                    )
+            create_insert_partition(table_path, df, partition_dict, storage_options)
+        elif mode == "merge":
+            # very slow -> https://github.com/delta-io/delta-rs/issues/1846
```

Comment:
This is not the case anymore :)

Reply:
Hi @ion-elgreco, thank you very much for all the comments; I am happy that there is somebody from the delta-rs side with some insights. I very much appreciate your help. 
I will recheck this when I finish general plugin refactoring on the dbt side and start again to test with the deltalake integration.